google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/TableAdminExample.java

```diff
@@ -0,0 +1,373 @@
+      return;
+    }
+    String projectId = args[0];
+    String instanceId = args[1];
+
+    TableAdminExample tableAdmin = new TableAdminExample(projectId, instanceId, "test-table");
+    tableAdmin.run();
+  }
+
+  public TableAdminExample(String projectId, String instanceId, String tableId) throws IOException {
+    this.tableId = tableId;
+
+    // [START connecting_to_bigtable]
+    // Creates the settings to configure a bigtable table admin client.
+    BigtableTableAdminSettings adminSettings =
+        BigtableTableAdminSettings.newBuilder()
+            .setProjectId(projectId)
+            .setInstanceId(instanceId)
+            .build();
+
+    // Creates a bigtable table admin client.
+    adminClient = BigtableTableAdminClient.create(adminSettings);
+    // [END connecting_to_bigtable]
+  }
+
+  public void run() {
+    createTable();
+    listAllTables();
+    getTableMeta();
+    maxAgeRule();
+    maxVersionsRule();
+    unionRule();
+    intersectionRule();
+    nestedRule();
+    listColumnFamilies();
+    modifyColumnFamilyRule();
+    printModifiedColumnFamily();
+    deleteColumnFamily();
+    deleteTable();
+    adminClient.close();
+  }
+
+  /** Demonstrates how to create a table with the specified configuration. */
+  public void createTable() {
+    // [START bigtable_create_table]
+    // Checks if table exists, creates table if does not exist.
+    if (!adminClient.exists(tableId)) {
+      System.out.println("Table does not exist, creating table: " + tableId);
+      CreateTableRequest createTableRequest = CreateTableRequest.of(tableId).addFamily("cf");
+      Table table = adminClient.createTable(createTableRequest);
+      System.out.printf("Table: %s created successfully%n", table.getId());
+    }
+    // [END bigtable_create_table]
+  }
+
+  /** Demonstrates how to list all tables within an instance. */
+  public void listAllTables() {
+    System.out.println("\nListing tables in current instance");
+    // [START bigtable_list_tables]
+    // Lists tables in the current instance.
+    try {
+      List<String> listTables = adminClient.listTables();
+      for (String tableName : listTables) {
+        System.out.println(tableName);
+      }
+    } catch (NotFoundException e) {
+      System.err.println("Failed to list tables from a non-existent instance: " + e.getMessage());
+    }
+    // [END bigtable_list_tables]
+  }
+
+  /** Demonstrates how to get a table's metadata. */
+  public void getTableMeta() {
+    System.out.println("\nPrinting table metadata");
+    // [START bigtable_get_table_metadata]
+    // Gets table metadata, and applies a view to the table fields.
+    try {
+      Table table = adminClient.getTable(tableId);
+      System.out.println("Table: " + table.getId());
+      Collection<ColumnFamily> columnFamilies = table.getColumnFamilies();
+      for (ColumnFamily columnFamily : columnFamilies) {
+        printColumnFamily(columnFamily);
```

Comment:
Although it will lead to a bit of duplication, I think it would be better to inline this function. I think the snippets should be mostly self contained